test/Rule.js

```diff
@@ -113,6 +113,106 @@ test('toConfig with values', t => {
   });
 });
 
+test('toConfig sorts before', t => {
+  const rule = new Rule();
+
+  rule
+    .use('babel')
+      .loader('babel-loader')
+      .options({ presets: ['alpha'] })
+      .end()
+    .use('alpha')
+      .loader('alpha-loader')
+      .end()
+    .use('cache')
+      .loader('cache-loader')
+      .before('babel');
+
+  t.deepEqual(rule.toConfig(), {
+    use: [
+      {
+        loader: 'cache-loader'
+      },
+      {
+        loader: 'babel-loader',
+        options: {
+          presets: ['alpha']
+        }
+      },
+      {
+        loader: 'alpha-loader'
+      }
+    ]
+  });
+});
+
+test('toConfig sorts after', t => {
+  const rule = new Rule();
+
+  rule
+    .use('babel')
+      .loader('babel-loader')
+      .options({ presets: ['alpha'] })
+      .end()
+    .use('alpha')
+      .loader('alpha-loader')
+      .end()
+    .use('cache')
+      .loader('cache-loader')
+      .after('babel');
+
+  t.deepEqual(rule.toConfig(), {
+    use: [
+      {
+        loader: 'babel-loader',
+        options: {
+          presets: ['alpha']
+        }
+      },
+      {
+        loader: 'cache-loader'
+      },
+      {
+        loader: 'alpha-loader'
+      }
+    ]
+  });
+});
+
+test('toConfig sorts before and after', t => {
+  const rule = new Rule();
+
+  rule
+    .use('babel')
+      .after('alpha')
+      .loader('babel-loader')
+      .options({ presets: ['alpha'] })
+      .end()
+    .use('alpha')
+      .loader('alpha-loader')
+      .end()
+    .use('cache')
+      .loader('cache-loader')
+      .before('babel');
+
+  t.deepEqual(rule.toConfig(), {
+    use: [
+      {
+        loader: 'alpha-loader'
+      },
+      {
+        loader: 'cache-loader'
+      },
+      {
+        loader: 'babel-loader',
+        options: {
+          presets: ['alpha']
+        }
+      }
+    ]
+  });
+});
+
```

Comment:
Could you add a test for the "before and after aren't allowed at the same time" handling?

I think we also need some tests for all of the merge omit functionality?